app/javascript/src/components/editor/TranslationKeys/TranslationKeysEditStrings.svelte

```diff
@@ -47,19 +67,26 @@
 
 <div class="sticky top-0 well well--dark block p-1/4 mb-1/4">
   <div class="flex gap-1/4 align-center">
-    <input class="form-input" value={selectedKey} bind:this={renameInput} on:input={renameKey} />
+    <input
+      class="form-input"
+      value={createNewKeyMode ? "" : selectedKey}
+      bind:this={createOrRenameInput}
+      on:input={!createNewKeyMode && debounce(createOrRenameKey, 300)}
+      on:change={createNewKeyMode && createOrRenameKey}/>
```

Comment:
Does it need to be on both input and change? That seems odd

Reply:
I was wondering if I should maybe add a comment here. I think for UX this is pretty nice:

- When creating a new key, the key won't be created until the user has pressed the Enter key (or moved focus away from the input)
- But when editing an already existing key, the explicit submission shouldn't be required (like with the translation \<textarea>s)

---

This is nice for users, but I agree its a bit confusing for developers. The main issue is I'm reusing the same input here for both. Maybe I should extract the input out, and make the different modes more separate in the code?

Reply:
Got it, that does make sense. Maybe there would be a way to do it both on input (or maybe keydown?) but extracted to a function somehow? Otherwise leaving a comment will do